src/mds/Server.h

```diff
@@ -327,6 +327,26 @@ class Server {
   bool terminating_sessions = false;
 
   std::set<client_t> client_reclaim_gather;
+  // record laggy sessions due to laggy OSDs
+  std::set<Session*> laggy_sessions;
+  int get_laggy_sessions() const {
+    return laggy_sessions.size();
+  }
+  void clear_laggy_sessions() {
+    if (laggy_sessions.size()) {
+      laggy_sessions.clear();
+    }
+  }
```

Comment:
we don't need these methods/member ^

Reply:
you mean remove `if` clause right? just like below

Reply:
i feel it's good to have it cause we need to get the size of set in beacon

Reply:
I'm saying there's no reason (AFAICT) to separate the two. Just use one set.

Reply:
More to the point: the distinction is not important to the admin. The warning should be clear that there is really one thing wrong: the MDS is not evicting clients as it normally would **because** one or more OSD are laggy. Why it would evict client X or Y is unimportant.

Reply:
got it, will have just one set, and its clear, get size helper.